app/assets/javascripts/rentals/new.js.erb

```diff
@@ -13,40 +13,47 @@ function days_between(sd, ed) {
   return Math.round(diff_ms/one_day) + 1;
 }
 
+function item_types_to_arr() {
+  //collect item types (will always be at least [])
+  return $(".item-type-quantity-picker").toArray().reduce(function (acc,part) {
+    var it_part = [];
+    for(var i=0;i<part.value;i++){
+      it_part.push($(part).attr("_name"));
+    }
+    return acc.concat(it_part);
+  }, []);
+}
+
 function costCalculation(){
-  var raw_item_type = null;
-  var raw_start_time = null;
-  var raw_end_time = null;
-  var amount = null;
-  var range = null;
+  //collect item types (will always be at least [])
+  var item_types = item_types_to_arr();
 
-  raw_item_type = $('#rental_item_type_id').val();
-  raw_start_time = $('#rental_start_time').val();
-  raw_end_time = $('#rental_end_time').val();
-  amount = $('#rental_amount');
-  date_range = $('#date_range');
+  var raw_start_time = $('#rental_start_time').val();
+  var raw_end_time = $('#rental_end_time').val();
+  var amount = $('#rental_amount');
+  var date_range = $('#date_range');
 
-  if(!isEmpty(raw_item_type) && !isEmpty(raw_end_time) && !isEmpty(raw_start_time)) {
+  if(!item_types.length <= 0 && !isEmpty(raw_end_time) && !isEmpty(raw_start_time)) {
     if(new Date(raw_start_time) > new Date(raw_end_time)) {
       return; //do not perform a cost calculation
       //the end_date changed to start_date + previous range then call this method once again
     }
-    range = days_between(raw_start_time, raw_end_time);
+    var range = days_between(raw_start_time, raw_end_time);
```

Comment:
Are you using Moment.js? If not I highly recommend it. 
http://momentjs.com/docs/

Reply:
I know we are using moment.js just not here, I could rewrite this method but id rather have it in a different pull request.  I was just adding the `var` keyword to keep the `range` variable within scope.